ndnsd/discovery/service-discovery.cpp

```diff
@@ -225,13 +211,11 @@ ServiceDiscovery::sendData(const ndn::Name& name)
   auto timeDiff = ndn::time::system_clock::now() - m_producerState.publishTimestamp;
   auto timeToExpire = ndn::time::duration_cast<ndn::time::seconds>(timeDiff);
 
-  int status = (timeToExpire > m_producerState.serviceLifetime) ? EXPIRED : ACTIVE;
+  m_serviceStatus = (timeToExpire > m_producerState.serviceLifetime) ? EXPIRED : ACTIVE;
```

Comment:
Move lines 211-214 to wireEncode too.
And you need to still have a local variable serviceStatus
and then check with global variable. Otherwise no need to wireEncode again.